gsy_framework/sim_results/carbon_emissions/results.py

```diff
@@ -42,31 +42,40 @@ def _find_start_and_end_dates(self, dates: List) -> Tuple[DateTime, DateTime]:
 
     def _create_hourly_timestamps(self, start: DateTime, end: DateTime) -> List[DateTime]:
         """Generate a list of datetime objects for each hour between start and end"""
+        if not isinstance(start, pendulum.DateTime):
+            start = pendulum.instance(start)
+        if not isinstance(end, pendulum.DateTime):
+            end = pendulum.instance(end)
         return [start.add(hours=hour) for hour in range(int((end - start).total_hours()) + 1)]
 
     def query_carbon_ratio_from_static_source(
-        self, country_code: str, start: DateTime, end: DateTime
+        self, country_code: str, start: DateTime, end: DateTime, file_prefix=""
```

Comment:
NIT: Let's also rename it here to
```suggestion
        self, country_code: str, start_time: DateTime, end_time: DateTime, file_prefix=""
```